x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_search_bar/url_synced_alerts_search_bar.tsx

```
@@ -0,0 +1,117 @@
+import { AlertsFeatureIdsFilter } from '../../lib/search_filters';
+import { useKibana } from '../../..';
+import { useAlertSearchBarStateContainer } from './use_alert_search_bar_state_container';
+import { ALERTS_URL_STORAGE_KEY } from './constants';
+import { AlertsSearchBarProps } from './types';
+import AlertsSearchBar from './alerts_search_bar';
+import { buildEsQuery } from '../global_alerts_page/global_alerts_page';
+import { nonNullable } from '../../../../common/utils';
+
+export type UrlSyncedAlertsSearchBarProps = Omit<
+  AlertsSearchBarProps,
+  'query' | 'rangeFrom' | 'rangeTo' | 'filters' | 'onQuerySubmit'
+> & {
+  onEsQueryChange: (esQuery: { bool: BoolQuery }) => void;
+  onActiveFeatureFiltersChange?: (value: AlertConsumers[]) => void;
+};
+
+/**
+ * An abstraction over AlertsSearchBar that syncs the query state with the url
+ */
+export const UrlSyncedAlertsSearchBar = ({
+  onEsQueryChange,
+  onActiveFeatureFiltersChange,
+  ...rest
+}: UrlSyncedAlertsSearchBarProps) => {
+  const {
+    data: { query: queryService },
+  } = useKibana().services;
+  const {
+    timefilter: { timefilter: timeFilterService },
+  } = queryService;
+  const {
+    kuery,
+    rangeFrom,
+    rangeTo,
+    filters,
+    onKueryChange,
+    onRangeFromChange,
+    onRangeToChange,
+    onFiltersChange,
+    savedQuery,
+    setSavedQuery,
+    clearSavedQuery,
+  } = useAlertSearchBarStateContainer(ALERTS_URL_STORAGE_KEY);
+
+  const syncEsQuery = useCallback(() => {
+    try {
+      onActiveFeatureFiltersChange?.([
+        ...new Set(
+          filters
+            .flatMap((f) => (f as AlertsFeatureIdsFilter).meta.alertsFeatureIds)
+            .filter(nonNullable)
+        ),
+      ]);
+      const newQuery = buildEsQuery({
+        timeRange: {
+          to: rangeTo,
+          from: rangeFrom,
+        },
+        kuery,
+        filters,
+      });
+      onEsQueryChange(newQuery);
+    } catch (e) {
+      // TODO show error message?
```

What causes this to fail? Why are we swallowing the error message without showing it? Not sure if this TODO should actually be a blocker.

---

I'm not entirely sure (never managed to break it through the KQL bar UI) but for good measure I've adopted the same solution as Observability (showing an error toast and resetting the query string)